### PR TITLE
feat(anta.tests): Added testcase to verify IP security connection

### LINKED ISSUE #567: Add support to validate IP sec connection status
Add support to validate IP sec connection status

```
wan1-dc1#show ip security connection 
Tunnel     Source           Dest             Status       Uptime     Input            Output           Rekey Time                       
path1      172.18.3.2       172.18.2.2       Established  2 hours    3187840 bytes    6155103 bytes    9 minutes   
                                                                     22560 pkts       47252 pkts                   
path2      172.18.3.2       172.18.1.2       Established  2 hours    2145680 bytes    5202796 bytes    9 minutes   
                                                                     14832 pkts       31792 pkts                   
path3      100.64.3.2       100.64.2.2       Idle         N/A        0 bytes          0 bytes          N/A         
                                                                     0 pkts           0 pkts                       
path4      100.64.3.2       100.64.1.2       Established  2 hours    3254348 bytes    6185981 bytes    11 minutes  
                                                                     22935 pkts       47998 pkts                   
path5      100.64.3.2       100.64.5.2       Established  2 hours    2130636 bytes    5191662 bytes    7 minutes   
                                                                     14670 pkts       31470 pkts                   
path6      172.18.3.2       172.18.5.2       Established  2 hours    2127016 bytes    5189950 bytes    8 minutes   
                                                                     14636 pkts       31406 pkts   
 ```

### PR DESCRIPTION
**Pull request by MaheshGSLAB**
# Description

Added testcase to verify IP security connection

Fixes #567

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have run pre-commit for code linting and typing (`pre-commit run`)
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes (`tox -e testenv`)

### CODE REVIEW

File: anta/tests/security.py [hunk 1]
```diff
@@ -512,3 +513,130 @@ def test(self) -> None:
...
+            if state != "Established":
+                failure_conn.append(connection)
+        if failure_conn:
+            failure_msg = "\n".join(failure_conn)
+            self.result.is_failure(f"Following IP security connections are not establised:\n{failure_msg}.")
+
+
+class VerifySpecificIPSecConn(AntaTest):
+    """
+    Verifies IP security connections for a peer.
```
Comment (MaheshGSLAB) [inline anta/tests/security.py:555]: Need to mention about only IPv4 support

File: anta/tests/security.py [hunk 2]
```diff
@@ -654,3 +655,151 @@ def test(self) -> None:
...
+                """Details of IPv4 security connections for a peer."""
+
+                source_address: IPv4Address
+                """Source address of the connection."""
+                destination_address: IPv4Address
+                """Destination address of the connection."""
+
+    def render(self, template: AntaTemplate) -> list[AntaCommand]:
+        """Render the template for each input IP Sec connection."""
+        return [template.render(peer=conn.peer, vrf=conn.vrf, connections=conn.connection) for conn in self.inputs.ip_security_connections]
```
Comment (gmuloc) [inline anta/tests/security.py:763]: so we want to stop overloading the template.params this way 
can you move the connection out and reparse the inputs in the test function?

**Review by gmuloc — CHANGES_REQUESTED** [review]

File: anta/tests/security.py (hunk 2 above)
Comment (MaheshGSLAB) [reply anta/tests/security.py]: updated in ba7f234

File: anta/tests/security.py [hunk 3]
```diff
@@ -654,3 +655,151 @@ def test(self) -> None:
...
+        # Check if IP security connection is configured
+        if not command_output:
+            self.result.is_failure("IPv4 security connection are not configured.")
+            return
+
+        # Iterate over all ip sec connection
+        for connection, conn_data in command_output.items():
+            state = next(iter(conn_data["pathDict"].values()))
+            if state != "Established":
+                failure_conn.append(connection)
```
Comment (gmuloc) [inline anta/tests/security.py:698]: Screenshot 2024-04-09 at 18 00 39
 This looks a bit confusing

Comment (gmuloc) [reply anta/tests/security.py:698]: maybe something like source: dest: vrf: ?

File: anta/tests/security.py [hunk 4]
```diff
@@ -654,3 +655,151 @@ def test(self) -> None:
...
+    @AntaTest.anta_test
+    def test(self) -> None:
+        """Main test function for VerifyIPSecConnHealth."""
+        self.result.is_success()
+        failure_conn = []
+        command_output = self.instance_commands[0].json_output["connections"]
+
+        # Check if IP security connection is configured
+        if not command_output:
+            self.result.is_failure("IPv4 security connection are not configured.")
```
Comment (gmuloc) [inline anta/tests/security.py:691]:
```suggestion
            self.result.is_failure("No IPv4 security connection configured.")
```

File: anta/tests/security.py [hunk 5]
```diff
@@ -654,3 +655,151 @@ def test(self) -> None:
...
+            return
+
+        # Iterate over all ip sec connection
+        for connection, conn_data in command_output.items():
+            state = next(iter(conn_data["pathDict"].values()))
+            if state != "Established":
+                failure_conn.append(connection)
+        if failure_conn:
+            failure_msg = "\n".join(failure_conn)
+            self.result.is_failure(f"Following IPv4 security connections are not establised:\n{failure_msg}.")
```
Comment (gmuloc) [inline anta/tests/security.py:701]:
```suggestion
            self.result.is_failure(f"The following IPv4 security connections are not establised:\n{failure_msg}.")
```

File: anta/tests/security.py [hunk 6]
```diff
@@ -654,3 +655,151 @@ def test(self) -> None:
...
+
+
+class VerifySpecificIPSecConn(AntaTest):
+    """
+    Verifies IPv4 security connections state for a peer.
```
Comment (gmuloc) [inline anta/tests/security.py:706]: Maybe mention that that some parameters are optionals

**Review by gmuloc — CHANGES_REQUESTED** [review]

**Review by gmuloc — COMMENTED** [review]
@MaheshGSLAB lets discuss tomorrow but I had to replace the `"null"` with `None` as this is what JSON `null` becomes in python.

**Review by gmuloc — APPROVED** [review]

**Pull request merged**